scripts/ledger_network_update.py

```diff
@@ -0,0 +1,398 @@
+
+        self.print_footer()
+
+    @staticmethod
+    def update_protobuf():
+        """Update protobuf dependency."""
+        click.echo("protobuf is not updating at the moment")
+
+    def update_spelling(self):
+        """Add network name to spelling file."""
+        click.echo("Add network name to spelling")
+        spelling = ROOT_DIR / ".spelling"
+        spelling.write_text(
+            spelling.read_text()
+            + f"\n{self.new_config.net_name}\n{self.new_config.chain_id}"
+        )
+
+    def update_cosmpy_version(self):
+        """Set new cosmpy version."""
+        click.echo("Update cosmpy version")
+        # pipenv
+        pipenv = ROOT_DIR / "Pipfile"
+        pipenv.write_text(
+            re.sub(
+                'cosmpy = ".*"', f'cosmpy = "{self.cosmpy_version}"', pipenv.read_text()
+            )
+        )
+        # aea ledger fetchai plugin
+        plugin_setup_py = ROOT_DIR / "plugins/aea-ledger-fetchai/setup.py"
+        plugin_setup_py.write_text(
+            re.sub(
+                '"cosmpy.*"',
+                f'"cosmpy{self.cosmpy_version}"',
+                plugin_setup_py.read_text(),
+            )
+        )
+
+        # aea ledger cosmos plugin
+        plugin_setup_py = ROOT_DIR / "plugins/aea-ledger-cosmos/setup.py"
+        plugin_setup_py.write_text(
+            re.sub(
+                '"cosmpy.*"',
+                f'"cosmpy{self.cosmpy_version}"',
+                plugin_setup_py.read_text(),
+            )
+        )
+
+        # tox
+        tox_file = ROOT_DIR / "tox.ini"
+        tox_file.write_text(
+            re.sub("cosmpy.*", f"cosmpy{self.cosmpy_version}", tox_file.read_text())
+        )
+
+    def update_aea_ledger_crypto(self):
+        """Update aea/ledger/crypto.py with new defaults."""
+        click.echo("Update aea/ledger/crypto.py")
+        content = AEA_LEDGER_MODULE_FILE.read_text()
+
+        content = content.replace(
+            f'FETCHAI_DEFAULT_ADDRESS = "{self.cur_config.rest_api_address}"',
+            f'FETCHAI_DEFAULT_ADDRESS = "{self.new_config.rest_api_address}"',
+        )
+        content = content.replace(
+            f'FETCHAI_DEFAULT_CURRENCY_DENOM = "{self.cur_config.denom}"',
+            f'FETCHAI_DEFAULT_CURRENCY_DENOM = "{self.new_config.denom}"',
+        )
+        content = content.replace(
+            f'FETCHAI_DEFAULT_CHAIN_ID = "{self.cur_config.chain_id}"',
+            f'FETCHAI_DEFAULT_CHAIN_ID = "{self.new_config.chain_id}"',
+        )
+
+        AEA_LEDGER_MODULE_FILE.write_text(content)
+
+    def update_docs(self):
+        """Update documentation."""
+        click.echo("Update docs")
+        docs_files = (ROOT_DIR / "docs").glob("**/*.md")
+        for f in docs_files:
+            content = f.read_text()
+            content = content.replace(
+                f"https://explore-{self.cur_config.net_name}.fetch.ai",
+                f"https://explore-{self.new_config.net_name}.fetch.ai",
```

Comment:
Let's take the whole explorer URL from input.

Reply:
@5A11 done